tests/integration/backend_test.py

```diff
@@ -969,9 +1011,232 @@ def test_qir_submission(authenticated_quum_backend: QuantinuumBackend) -> None:
+  br label %continue
+
+continue:                                         ; preds = %else, %then
+  call void @__quantum__qis__h__body(%Qubit* null)
+  call void @__quantum__rt__tuple_start_record_output()
+  %8 = call i64 @get_int_from_creg(i1* %0)
+  call void @__quantum__rt__int_record_output(i64 %8, i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0))
+  %9 = call i64 @get_int_from_creg(i1* %1)
+  call void @__quantum__rt__int_record_output(i64 %9, i8* getelementptr inbounds ([5 x i8], [5 x i8]* @1, i32 0, i32 0))
+  %10 = call i64 @get_int_from_creg(i1* %2)
+  call void @__quantum__rt__int_record_output(i64 %10, i8* getelementptr inbounds ([5 x i8], [5 x i8]* @2, i32 0, i32 0))
+  %11 = call i64 @get_int_from_creg(i1* %3)
+  call void @__quantum__rt__int_record_output(i64 %11, i8* getelementptr inbounds ([5 x i8], [5 x i8]* @3, i32 0, i32 0))
+  call void @__quantum__rt__tuple_end_record_output()
+  ret void
+}
+
+declare i1 @get_creg_bit(i1*, i64)
+
+declare void @set_creg_bit(i1*, i64, i1)
+
+declare void @set_creg_to_int(i1*, i64)
+
+declare i1 @__quantum__qis__read_result__body(%Result*)
+
+declare i1* @create_creg(i64)
+
+declare i64 @get_int_from_creg(i1*)
+
+declare void @mz_to_creg_bit(%Qubit*, i1*, i64)
+
+declare void @__quantum__rt__int_record_output(i64, i8*)
+
+declare void @__quantum__rt__tuple_start_record_output()
+
+declare void @__quantum__rt__tuple_end_record_output()
+
+declare void @__quantum__qis__x__body(%Qubit*)
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__z__body(%Qubit*)
+
+attributes #0 = { "EntryPoint" "maxQubitIndex"="1" "maxResultIndex"="1" "requiredQubits"="3" "requiredResults"="3" }
+
+!llvm.module.flags = !{!0, !1, !2, !3}
+
+!0 = !{i32 1, !"qir_major_version", i32 1}
+!1 = !{i32 7, !"qir_minor_version", i32 0}
+!2 = !{i32 1, !"dynamic_qubit_management", i1 false}
+!3 = !{i32 1, !"dynamic_result_management", i1 false}
+
+"""
+    ctx = create_context()
+    module = parse_assembly(qir, context=ctx)
+    ir = module.as_bitcode()
+    h = b.submit_program(Language.QIR, b64encode(ir).decode("utf-8"), n_shots=10)
+
+    with pytest.raises(GetResultFailed):
+        r = b.get_result(h)
+
+    assert "Unexpected Instruction   call void @mz_to_creg_bit" in str(
+        b.circuit_status(h)
+    )
+    # if this checks fails try to solve
+    # https://github.com/CQCL/pytket-quantinuum/issues/358
+
+
+@pytest.mark.skipif(skip_remote_tests, reason=REASON)
+@pytest.mark.parametrize(
+    "authenticated_quum_backend_qa", [{"device_name": "H1-1SC"}], indirect=True
+)
+@pytest.mark.timeout(120)
+def test_qir_submission_mz_to_reg_qa(
+    authenticated_quum_backend_qa: QuantinuumBackend,
+) -> None:
+    # disable Garbage Collector because of
+    # https://github.com/CQCL/pytket-quantinuum/issues/170
+    gc.disable()
+    b = authenticated_quum_backend_qa
+    qir = """; ModuleID = 'test_pytket_qir_6'
+source_filename = "test_pytket_qir_6"
```

Comment:
Maybe these big QIR programs could be read from files rather than the test source.

Reply:
I have moved all the qir to external files